### DISCUSSION TITLE
Pnp Search v4 search box not showing result if column value is a phrase

### DISCUSSION THREAD

**Discussion by Emmanueladegor** [microsoft-search/pnp-modern-search#2965] [Q&A] [answered]
Hello all, 

I have everything mapped correctly from my SP library to Managed properties. When I search on the search box, all column search works and returns the result if the column contains a single value (ex: 7803 or "part"). If it contains words like a single line of text "Document name" column (ex: "party leader list.pdf"), Pnp search v4 shows zero results when this value is searched.

My query template is like this:

RefinableString00: {searchTerm} or
RefinableString01: {searchTerm} or
RefinableString02: {searchTerm} or
RefinableString03: {searchTerm} or
RefinableString04: {searchTerm}

00, 01, 02, and 04 works because the document library columns contain a single value but RefinableString03 is mapped to a column with words that contain spaces like a phrase.

How do I make this work? I tried adding wildcards like so: RefinableString03: *{searchTerm}*, but still returned no results.

Thank You 

 @wobba @stevebeauge @mrjarbenne @PooLP @edarroudi

**Comment by stevebeauge** [comment]
If you switch to debug layout, you'll see a `QueryModification` somewhere in the response. This should contains the actual query passed to the search engine after the tokens are resolved. What is the value of this property ?

If you manually enter the term with something like `RefinableString03:"party leader list.pdf"` does it return the expected results ?

Using the search query tool, using the query : `path:"https://yourtenant.sharepoint.com/sites/somesites/document/party leader list.pdf`, do you see the proper `RefinableString03` value ?

**Reply by Emmanueladegor** [reply]
Hello @stevebeauge , 

Thanks for responding.

If I do RefinableString03:"party leader list.pdf", it works.
If I do RefinableString03:"{searhTerms}", it works.

But when I do:
RefinableString00:{searhTerms} OR
RefinableString01:{searhTerms} OR
RefinableString02:{searhTerms} OR
RefinableString03:"{searhTerms}" OR
RefinableString04:{searhTerms} 

Then QueryModification shows:

"RefinableString00: party leader list.pdf"

I think that's why it doesn't search. It's because I add the other refinables. 

Please advise me on how to structure this

Thank you

**Reply by Emmanueladegor** [reply]
@stevebeauge - My temporary fix is to reorder the query by doing this:

RefinableString03:"{searhTerms}" OR
RefinableString00:{searhTerms} OR
RefinableString01:{searhTerms} OR
RefinableString02:{searhTerms} OR
RefinableString04:{searhTerms}

This works! But I'll like to know if there's a better method.

Thank you

**Comment by kasperbolarsen** [comment]
Hi , is the query exactly as shown here? I am pretty sure you will need to have each {searchTerms} in quotes like (and wrap it in () if you have other elements in you query):  

(RefinableString03:"{searhTerms}" OR
RefinableString00:"{searchTerms}" OR
RefinableString01:"{searchTerms}" OR
RefinableString02:"{searchTerms}" OR
RefinableString04:"{searchTerms}")

**Reply by Emmanueladegor** [reply]
Got it! 

Thank you all, my issue is sorted now. We can close the thread.

**Comment by wobba** [comment, marked as answer]
Here's the correct way to make it parse like you want. See https://learn.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#grouping-property-restrictions-within-a-kql-query

```
RefinableString00:({searchTerm}) OR
RefinableString01:({searchTerm}) OR
RefinableString02:({searchTerm}) OR
RefinableString03:({searchTerm}) OR
RefinableString04:({searchTerm})
```

**Reply by Emmanueladegor** [reply]
@wobba - Thank you! You're awesome!
